### ISSUE TITLE
Error when selecting not existing combination with redirectOnPropertyChange enabled

### ISSUE THREAD

**Issue by Geoffry304** [OFFLINE-GmbH/oc-mall-plugin#485]
Hello,

First of all thank you for this great plugin!

I was wondering if it is possible to show only the correct property values for the selected "Attribute for variant grouping"

For example: I have a product : Phone A and that product has multiple variants and I am filtering on color for 'Attribute for variant grouping':
Name|Color|memory|grade
Phone A | Color White | 1gb | grade A
Phone A | Color White | 1gb | grade B
Phone B | Color Black | 1gb | grade A
Phone B | Color White | 2gb | grade A

If I would now go to the product detail page of the first phone, it will also give me the option to select a memory of 2gb and this will result in an error. 
So is it possible to only get the available values for the 'Attribute for variant grouping' ?

I hope you understand my issue, if not please tell me and I try to explain it better!

![image](https://user-images.githubusercontent.com/3656871/78609639-6207df80-7863-11ea-97f8-f01dcf48b7b6.png)

As you can see in the image above the blue one has no C grade but I can select it.

![image](https://user-images.githubusercontent.com/3656871/78609700-8a8fd980-7863-11ea-951a-4e4c644f1b1a.png)

Thanks in advance,
Geoffry

**Comment by tobias-kuendig** [comment]
We currently do not check if each variant combination is available. 

We would need to check if `$item` is null here:
https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/components/Product.php#L358

If it is `null` a message would have to be returned to the user that the selected variant does not exist. 

A better solution would be to only show possible combinations but this requires us to update the dropdowns after each selection and might need some thinking to get it right.

Also, if you disable the `redirectOnPropertyChange` feature, the selected combination will be displayed as `out of stock`. So this might be a simple workaround for you for this problem.

**Comment by Geoffry304** [comment]
Thanks for the response, I know when i disable the redirectOnPropertyChange it is fixed but then only the stock and price updates, not the product info, so if I select another grade and refresh the page it will go back to the old grade. 

It would indeed be  a better solution to only show the possible combination but when you put on the redirectOnPropertyChange after each selection the page will refresh so the dropdowns will be automatically updated?
It would also be ok if  the property is displayed but disabled when it doesn't exist.
I will also look into this!

**Comment by damsfx** [comment]
> 
> 
> Also, if you disable the `redirectOnPropertyChange` feature, the selected combination will be displayed as `out of stock`. So this might be a simple workaround for you for this problem.

In this case, you can't get per varaint description and title to be displayed in the page!?

**Comment by damsfx** [comment]
> 
> 
> ...  when i disable the redirectOnPropertyChange it is fixed but then only the stock and price updates, not the product info ...

@tobias-kuendig  
I get the possibility for the page to refresh the "infos" section by modifying the returned array for the `stockCheckResponse()` function of `Product` component at line [710](https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/e7055e48ab6b1c03a178bf236c0e17be4f972903/components/Product.php#L710) : 
from
````
        return [
            '.mall-product__price'       => $this->renderPartial($this->alias . '::price', $data),
            '.mall-product__add-to-cart' => $this->renderPartial($this->alias . '::addtocart', $data),
        ];
```` 
to
````
        return [
            '.mall-product__info'        => $this->renderPartial($this->alias . '::info', $data),
            '.mall-product__price'       => $this->renderPartial($this->alias . '::price', $data),
            '.mall-product__add-to-cart' => $this->renderPartial($this->alias . '::addtocart', $data),
        ];
```` 
It remains consistent with your theme.

**Comment by tobias-kuendig** [comment]
@damsfx FYI this change causes issues with the ProductReviews component (see #548). A fix was added in 1ccae01.

**Comment by tobias-kuendig** [comment]
I'm closing this issue due to inactivity. If you need any further assistance feel free to reply.
